code/modules/food_and_drinks/kitchen_machinery/food_replicator.dm

```diff
@@ -0,0 +1,631 @@
+	handle_emag_effect(poly)
+	sleep(20)
+	adjust_paste(-sides)
+	if(!silent)		//won't use power when we spit out excess (since that runs silently)
+		use_power(active_power_usage)
+	poly.forceMove(get_turf(src))
+	operating = 0
+	if(!silent)
+		playsound(loc, 'sound/machines/ding.ogg', 50, 1)
+	return 1
+
+//adjust_paste is used to alter and sanity check the food_paste value.
+//if food_paste is above the max_food_paste, it will begin printing off a pile of food polys until it is at or below max again.
+/obj/machinery/food_replicator/proc/adjust_paste(amount = 0)
+	food_paste = max(0, (food_paste + amount))
+
+	if(food_paste > max_food_paste)
+		var/paste_to_waste = food_paste - max_food_paste
+		if(paste_to_waste >= 20)
+			print_poly(20, 1)
+		else if(paste_to_waste >= 12)
+			print_poly(12, 1)
+		else if(paste_to_waste >= 8)
+			print_poly(8, 1)
+		else if(paste_to_waste >= 6)
+			print_poly(6, 1)
+		else if(paste_to_waste >= 4)
+			print_poly(4, 1)
+		else
+			print_poly(1, 1)
+
+/obj/machinery/food_replicator/attack_ai(mob/user)
+	return attack_hand(user)
+
+/obj/machinery/food_replicator/attack_hand(mob/user)
+	if(stat & (BROKEN|NOPOWER))
+		return
+	user.set_machine(src)
+	ui_interact(user)
+
+/obj/machinery/food_replicator/interact(mob/user)
+	if(panel_open || !anchored)
+		return
+	//todo
+
+/obj/machinery/food_replicator/attackby(obj/item/O, mob/user, params)
+	if(operating)
+		return
+	if(default_deconstruction_screwdriver(user, open_icon, off_icon, O))
+		return
+	if(default_unfasten_wrench(user, O))
+		return
+	if(exchange_parts(user, O))
+		return
+	default_deconstruction_crowbar(O)
+
+	if(istype(O, /obj/item/weapon/reagent_containers/food/snacks))
+		if(loaded)
+			to_chat(user, "<span class='warning'>There is already a dish loaded. Recycle or eject it before adding a new dish.</span>")
+			return
+		var/obj/item/weapon/reagent_containers/food/snacks/to_load = O
+		if(!user.canUnEquip(to_load, 0))
+			to_chat(user, "<span class='warning'>[to_load] is stuck to your hand, you can't seem to put it down!</span>")
+			return
+		user.unEquip(to_load)
+		to_load.forceMove(src)
+		loaded = to_load
+		nanomanager.update_uis(src)
+		return 1
+
+	if(panel_open && iswire(O))
+		var/obj/item/stack/cable_coil/C = O
+		if(stat & BROKEN)
+			if(C.amount >= 5)
+				if(do_after(user, 40, target = src))
+					to_chat(user, "<span class='notice'>You replace the burnt out wiring, restoring functionality.</span>")
+					C.use(5)
+					set_fix()
+			else
+				to_chat(user, "<span class='warning'>You don't have enough cable to repair the wiring.</span>")
+		else
+			to_chat(user, "<span class='notice'>The wiring doesn't require replacing at this time.</span>")
+		return 1
+
+	if(istype(O, /obj/item/weapon/card/emag))
+		emag_act(user)
+		return 1
+
+/obj/machinery/food_replicator/emag_act(mob/user)
+	if(!emagged)
+		to_chat(user, "<span class='warning'>You corrupt the reagent replication subroutine to lace created dishes with poisons.</span>")
```

Comment:
I mean, I get that this is a cool idea. But most other emag things do stuff simple like tripping the safeties... It triggering poison instead of the nutriment etc... seems far too intricate and involved for a simple emag operation.